test/success/hlevels.prl

```diff
@@ -0,0 +1,74 @@
+Def HasAllPathsTo (#C,#c) = [(-> [c' : #C] (path [_] #C c' #c))].
+Def IsContr (#C) = [(* [c : #C] (HasAllPathsTo #C c))].
+Def IsProp (#A) = [(-> [a b : #A] (path [_] #A a b))].
+Def IsSet (#A) = [(-> [a b : #A] (IsProp (path [_] #A a b)))].
+
+Thm InhPropIsContr(#l:lvl) : [
+ (->
+  [ty : (U #l kan)] // can I use pre here?
+  [h : (IsProp ty)]
+  [a : ty]
+  (IsContr ty))
+] by [
+  unfold IsProp;
+  lam ty, h, a. {use a,lam x. `($ h x a)}
+].
+
+Thm PropPi(#l:lvl) : [
+ (->
+  [tyA : (U #l kan)]
+  [tyB : (-> tyA (U #l kan))]
+  [h : (-> [x : tyA] (IsProp ($ tyB x)))]
+  (IsProp (-> [x : tyA] ($ tyB x))))
+] by [
+  unfold IsProp; lam tyA, tyB, h, f, g.
+  <i> lam x. `(@ ($ h x ($ f x) ($ g x)) i)
+].
+
+Thm PropSet(#l:lvl) : [
+  (->
+   [tyA : (U #l kan)]
+   [h : (IsProp tyA)]
+   (IsSet tyA)
+  )
+] by [
+  // unfold IsSet, IsProp; // why can't I chain unfolds like I can for lam?
+  unfold IsSet; unfold IsProp; // why do I need a ; and not .?
```

Comment:
The `;` is a tactical that sequences tactics, like in Coq. The `.` doesn't mean anything on its own, but is part of the notation for certain tacticals, including `lam x. tac`.

Reply:
This is extremely confusing if you know Coq... Why not use `.` to separate tactics and `;` for sequencing them?

Reply:
We don't have any 'top-level' interact like in Coq; what I mean is that in Coq, you're focused on the first subgoal, and `.` just applies a tactic to that subgoal. RedPRL is more in the style of LCF, where you must write a single script to cover your goals, rather than issue commands that change the proof state (as in coq). So, we use the period for a different purpose.

Since we don't have anything analogous to the `.`-structured proofs of Coq, I think there is no opportunity for confusion.

Reply:
We might choose a different delimiter than a period, of course... In some cases, something like `=>` might work nicely; on the other hand, it wouldn't work for the `let` expressions.

anyway, I'm open to designing a better concrete syntax, since the current one is pretty crappy.

Reply:
Oh, I understand now! I'm not used to writing a single tactic to solve all subgoals... I can't wait for the RedPRL tutorial so that I can learn what is going on without having to ask a million stupid questions :)